follow_sponsorship/workflow/contract_workflow.xml

```diff
@@ -0,0 +1,162 @@
+        </record>
+        
+        <record model="workflow.activity" id="act_no_sub">
+            <field name="wkf_id" ref="recurring_contract.wkf_recurring_contract" />
+            <field name="name">no_sub</field>
+            <field name="kind">function</field>
+            <field name="action">no_sub()</field>
+        </record>
+        
+        <record model="workflow.activity" id="act_sub">
+            <field name="wkf_id" ref="recurring_contract.wkf_recurring_contract" />
+            <field name="name">sub</field>
+            <field name="kind">function</field>
+            <field name="action">sub()</field>
+        </record>
+        
+        <record model="workflow.activity" id="act_sub_accept">
+            <field name="wkf_id" ref="recurring_contract.wkf_recurring_contract" />
+            <field name="name">sub_accept</field>
+            <field name="kind">function</field>
+            <field name="action">sub_accept()</field>
+        </record>
+        
+        <record model="workflow.activity" id="act_sub_reject">
+            <field name="wkf_id" ref="recurring_contract.wkf_recurring_contract" />
+            <field name="name">sub_reject</field>
+            <field name="kind">function</field>
+            <field name="action">sub_reject()</field>
+        </record>
+
+        <!-- Transitions -->
+        <record model="workflow.transition" id="t0">
+            <field name="act_from" ref="act_draft" />
+            <field name="act_to" ref="act_start" />
+            <field name="signal">contract_validated</field>
+        </record>
+
+        <record model="workflow.transition" id="t1a">
+            <field name="act_from" ref="act_start" />
+            <field name="act_to" ref="act_waiting_welcome" />
+            <field name="signal">mail_sent</field>
+            <field name="condition">correspondant_id.email</field>
+        </record>
+
+         <record model="workflow.transition" id="t1b">
+            <field name="act_from" ref="act_start" />
+            <field name="act_to" ref="act_active" />
+            <field name="signal">mail_sent</field>
+        </record>
+        
+        <record model="workflow.transition" id="t2">
+            <field name="act_from" ref="act_waiting_welcome" />
+            <field name="act_to" ref="act_active" />
+            <field name="signal">welcome_sent</field>
+        </record>
+        
+        <record model="workflow.transition" id="t5a">
+            <field name="act_from" ref="act_active" />
+            <field name="act_to" ref="act_sub_waiting" />
+            <field name="signal">contract_terminated</field>
+            <field name="condition">end_reason=='1'</field>
+        </record>
+        
+        <record model="workflow.transition" id="t5b">
+            <field name="act_from" ref="act_active" />
+            <field name="act_to" ref="act_cancelled" />
+            <field name="signal">contract_terminated</field>

```

Comment:
add condition end_reason != '1' to be sure it will not be taken in case it were evaluated before transition t5a